Numbers/PawnColumnWorkers/PawnColumnWorker_Entropy.cs

```diff
@@ -0,0 +1,48 @@
+﻿namespace Numbers
+{
+    using System.Collections.Generic;
+    using System.Reflection;
+    using RimWorld;
+    using UnityEngine;
+    using Verse;
+
+    [StaticConstructorOnStartup]
+    public class PawnColumnWorker_Entropy : PawnColumnWorker
+    {
+        private static readonly Texture2D EntropyBarTex = SolidColorMaterials.NewSolidColorTexture(new Color(0.46f, 0.34f, 0.35f));
+
+        //mostly from PawnColumnWorker_Need
+
+        public override void DoCell(Rect rect, Pawn pawn, PawnTable table)
+        {
+            if (!pawn.HasPsylink)
+                return;
+
+            float curEntropyLevel = pawn.psychicEntropy.EntropyRelativeValue;
+
+            float barHeight = 14f;
+            float barWidth = barHeight + 15f;
+            if (rect.height < 50f)
+            {
+                barHeight *= Mathf.InverseLerp(0f, 50f, rect.height);
+            }
+
+            Text.Font = (rect.height <= 55f) ? GameFont.Tiny : GameFont.Small;
+            Text.Anchor = TextAnchor.UpperLeft;
+            Rect rect3 = new Rect(rect.x, rect.y + rect.height / 2f, rect.width, rect.height / 2f);
+            rect3 = new Rect(rect3.x + barWidth, rect3.y, rect3.width - barWidth * 2f, rect3.height - barHeight);
+
+            Widgets.FillableBar(rect3, curEntropyLevel, EntropyBarTex);
+
+            Text.Font = GameFont.Small;
+        }
+
+        public override int GetMinWidth(PawnTable table)
+            => Mathf.Max(base.GetMinWidth(table), 110);
+
+        public override int Compare(Pawn a, Pawn b)
+        {
+            return ((float)(a.psychicEntropy?.EntropyRelativeValue ?? -1.0f)).CompareTo((float)(b.psychicEntropy?.EntropyRelativeValue ?? -1.0f));
```

Comment:
This doesn't work at all! I have no idea!

Reply:
Also applies to https://github.com/Mehni/kNumbers/pull/29/files#r540718444

Reply:
Top two guesses on the board:

1. There's a really dumb typo in these 2 lines that I'm just not seeing. I mangled them up a bunch (added brackets, typecasts, etc) and nothing helped, though.
2. The sort is assuming this operator is transitive, when it is not. Just because pawns without psylinks are sorting equal to each other, doesn't mean they sort correctly against pawns with psylinks.

Reply:
You can simplify it by removing the cast.

`return (a.psychicEntropy?.EntropyRelativeValue ?? -1.0f).CompareTo(b.psychicEntropy?.EntropyRelativeValue ?? -1.0f);`

but the code looks fine to me. I'd expect pawns with psychicEntropy to be grouped together, same as pawns without.

Reply:
I figured it out. The code was fine as original (and all iterations on it). Rather, the issue was assumptions with game logic. 

Pawns with and without Psylinks can both clock the same Entropy of 0, and so a bunch of Heat bars at 0 were staying scattered, and not clustering, as you'd expect.

Pawns without Psylinks for some reason have a totally arbitrary and random-seeming amount of Psyfocus, even though you can't see it anywhere outside the Debug Inspector and I don't think it does anything.

Now, for both stats, we first compare on whether they have a Psylink, before anything else.

Now, pawns with bars now correctly cluster together to differentiate from pawns without bars at all, and then we sort by value.